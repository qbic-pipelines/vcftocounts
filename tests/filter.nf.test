nextflow_pipeline {

    name "Test pipeline with filter settings"
    script "../main.nf"

    test("Params: filter | input filter test") {

        config 'conf/test_filter.config'

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // vcf_files: All files in ${params.outdir}
            def vcf_files  = getAllFilesFromDir(params.outdir,
                include: ['**/*.vcf.gz', '**/*.vcf'],
                ignore: ['**/concat.concat.vcf.gz',
                    '**/toindex_concat_merge.merge.vcf.gz',
                    '**/concat.reheader.vcf.gz'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/vcftocounts_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All vcf files
                    vcf_files.collect{ file -> file.name + ":md5," + path(file.path).vcf.variantsMD5 }
                ).match() }
            )
        }
    }

    test("Params: filter | input filter test - stub") {

        options "-stub"

        config 'conf/test_filter.config'

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // vcf_files: All files in ${params.outdir}
            def vcf_files  = getAllFilesFromDir(params.outdir, include: ['**/*.vcf.gz', '**/*.vcf'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/vcftocounts_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All vcf files
                    vcf_files
                ).match() }
            )
        }
    }
}
