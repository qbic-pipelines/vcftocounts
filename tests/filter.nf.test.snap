{
    "Params: filter | input filter test": {
        "content": [
            8,
            {
                "BCFTOOLS_ANNOTATE": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_REHEADER": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_VIEW": {
                    "bcftools": 1.2
                },
                "CREATE_SAMPLE_FILE": {
                    "create_sample_file": "1.0.0"
                },
                "TABIX_TABIX": {
                    "tabix": 1.2
                },
                "VCF2COUNTS": {
                    "vcf2counts.R": "1.0.0"
                },
                "Workflow": {
                    "qbic-pipelines/vcftocounts": "v2.0.1dev"
                }
            },
            [
                "bcftools",
                "bcftools/annotate",
                "bcftools/annotate/chr21_ann.reheader.vcf.gz.csi",
                "bcftools/annotate/chr21_ann_full.IDremoved.vcf.gz",
                "bcftools/annotate/versions.yml",
                "bcftools/reheader",
                "bcftools/reheader/chr21_ann.reheader.vcf.gz",
                "bcftools/reheader/chr21_ann.reheader.vcf.gz.tbi",
                "bcftools/reheader/samplefiles",
                "bcftools/reheader/samplefiles/chr21_ann.txt",
                "bcftools/reheader/samplefiles/versions.yml",
                "bcftools/reheader/versions.yml",
                "bcftools/view",
                "bcftools/view/chr21_ann_full.filter.vcf.gz",
                "bcftools/view/chr21_ann_full.filter.vcf.gz.tbi",
                "bcftools/view/versions.yml",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/vcftocounts_software_mqc_versions.yml",
                "tabix",
                "tabix/annotated.vcf.gz.tbi",
                "vcf2counts",
                "vcf2counts/chr21_ann_full.csv"
            ],
            [
                "versions.yml:md5,74b0a613672a9941d2bb2a62c18406bb",
                "chr21_ann.txt:md5,67a48a0a8902f9e8493f85a6de469ebb",
                "versions.yml:md5,4afe7b519f82d5d8c539950934477c1a",
                "versions.yml:md5,e516c8ae57ef27b417031a3d10715b00",
                "versions.yml:md5,3d4011c68f8fe5b9c63cc55f23613599",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "chr21_ann_full.csv:md5,e797160c3cae2431c11d0db9208555f9"
            ],
            [
                "chr21_ann_full.IDremoved.vcf.gz:md5,188c2be2f427d894e8bbf32911dcdb7e",
                "chr21_ann.reheader.vcf.gz:md5,c507b1cbd540e8324f686618013ad137",
                "chr21_ann_full.filter.vcf.gz:md5,97cd79b71fb0424bfa642e23f22bb035"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-10T10:54:30.320115"
    },
    "Params: filter | input filter test - stub": {
        "content": [
            8,
            {
                "BCFTOOLS_ANNOTATE": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_REHEADER": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_VIEW": {
                    "bcftools": 1.2
                },
                "CREATE_SAMPLE_FILE": {
                    "create_sample_file": "1.0.0"
                },
                "TABIX_TABIX": {
                    "tabix": 1.2
                },
                "VCF2COUNTS": {
                    "vcf2counts.R": "1.0.0"
                },
                "Workflow": {
                    "qbic-pipelines/vcftocounts": "v2.0.1dev"
                }
            },
            [
                "bcftools",
                "bcftools/annotate",
                "bcftools/annotate/chr21_ann_full.IDremoved.vcf.gz",
                "bcftools/annotate/versions.yml",
                "bcftools/reheader",
                "bcftools/reheader/chr21_ann.reheader.vcf.gz",
                "bcftools/reheader/chr21_ann.reheader.vcf.gz.tbi",
                "bcftools/reheader/samplefiles",
                "bcftools/reheader/samplefiles/chr21_ann.txt",
                "bcftools/reheader/samplefiles/versions.yml",
                "bcftools/reheader/versions.yml",
                "bcftools/view",
                "bcftools/view/chr21_ann_full.filter.vcf.gz",
                "bcftools/view/chr21_ann_full.filter.vcf.gz.tbi",
                "bcftools/view/versions.yml",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/vcftocounts_software_mqc_versions.yml",
                "tabix",
                "tabix/annotated.vcf.gz.csi",
                "tabix/annotated.vcf.gz.tbi",
                "vcf2counts",
                "vcf2counts/chr21_ann_full.csv"
            ],
            [
                "versions.yml:md5,74b0a613672a9941d2bb2a62c18406bb",
                "chr21_ann.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "versions.yml:md5,4afe7b519f82d5d8c539950934477c1a",
                "versions.yml:md5,e516c8ae57ef27b417031a3d10715b00",
                "versions.yml:md5,3d4011c68f8fe5b9c63cc55f23613599",
                "chr21_ann_full.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                "chr21_ann_full.IDremoved.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "chr21_ann.reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "chr21_ann_full.filter.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-10T10:54:55.972419"
    }
}